src/server/object_services/schemas/object_md_indexes.js

```diff
@@ -8,12 +8,10 @@ module.exports = [
     {
         // find_object_latest()
         // list_objects()
+        // note that version_past is always null here by partialFilterExpression.
         fields: {
             bucket: 1,
             key: 1,
-            // we include version_past as extra index field to separate from null_version_index.
-            // note that version_past is always null here by partialFilterExpression.
-            version_past: 1,
```

Comment:
I think we need the same handling for the other two indexes - latest_version_index and null_version_index where we also used a boolean field which is null by default

Reply:
let's do it in a different PR so we will keep PRs small

Reply:
I created another PR that fixes null_version_index